test/hotspot/jtreg/serviceability/monitoring/ThreadInfo/getLockOwnerName/getLockOwnerName.java

```diff
@@ -0,0 +1,415 @@
+                                           + "returned null name for "
+                                           + "contender.");
+            } else if (!name.equals("blocker")) {
+                out.println("Failure at " + count + " loops.");
+                throw new RuntimeException("name='" + name + "': name "
+                                           + "should be blocker.");
+            } else {
+                logDebug("ThreadInfo.getLockOwnerName() returned blocker.");
+            }
+
+            synchronized (barrierReleaser) {
+                // tell releaser thread to release the blocker thread
+                testState = TS_READY_TO_RELEASE;
+                barrierReleaser.notify();
+            }
+
+            while (true) {
+                // maxDepth == 0 requires no safepoint so alternate.
+                int maxDepth = ((count % 1) == 1) ? Integer.MAX_VALUE : 0;
+                info = mbean.getThreadInfo(id, maxDepth);
+                if (info == null) {
+                    // the contender has exited
+                    break;
+                }
+                name = info.getLockOwnerName();
+                // We can't verify that name == null here because contender
+                // might be slow leaving the threadLock monitor.
+                lateCount++;
+            }
+            logDebug("made " + lateCount + " late calls to getThreadInfo() " +
+                     "and info.getLockOwnerName().");
+
+            try {
+                releaser.join(JOIN_MAX * 1000);
+                if (releaser.isAlive()) {
+                    System.err.println("Failure at " + count + " loops.");
+                    throw new InternalError("releaser thread is stuck");
+                }
+                blocker.join(JOIN_MAX * 1000);
+                if (blocker.isAlive()) {
+                    System.err.println("Failure at " + count + " loops.");
+                    throw new InternalError("blocker thread is stuck");
+                }
+                contender.join(JOIN_MAX * 1000);
+                if (contender.isAlive()) {
+                    System.err.println("Failure at " + count + " loops.");
+                    throw new InternalError("contender thread is stuck");
+                }
+            } catch (InterruptedException ex) {
+            }
+
+            checkTestState(TS_CONTENDER_DONE);
+        }
+
+        System.out.println("Executed " + count + " loops in " + timeMax +
+                           " seconds.");
+
+        return 0;
+    }
+}
+
+class getLockOwnerNameWorker extends Thread {
+    public getLockOwnerNameWorker(String name) {
+        super(name);
+    }
+
+    public void run() {
+        getLockOwnerName.logDebug("thread running");
+
+        //
+        // Launch the blocker thread:
+        // - grabs threadLock
+        // - holds threadLock until we tell it let go
+        // - releases threadLock
+        //
+        if (getName().equals("blocker")) {
```

Comment:
Bit of a nit here, but is there a reason not to just create separate Thread subclasses for each of the 3 types of threads you are handling here? Or just make these each separate static methods of the main class and use the instantiate the Thread class with a lambda.

Reply:
This new test is a variation of a 20 year old test that I recently ported to JVM/TI
and integrated. 20 years ago, it was much simpler to write the test this way.
I could create a separate Thread subclass for each "role", but I'd rather not
do that since it will no longer be easy to compare this test to its siblings.

As for lambdas, I know absolutely zero about writing lambda code.

Reply:
Ok. I get it about lambdas, but they can be useful for simplifying thread tasks without creating a subclass. Here are a few examples, but no need for you to replicate any them:

```
    // create thread with specified method as the "run" method
    Thread t = new Thread(this::testMethod);
    t.start();
```

```
    // create thread with the specified code block as the "run" method
    Thread t1 = new Thread(() -> {
        synchronized (lock1) {
            System.out.println("Thread in sync section 1: "
                    + Thread.currentThread().getName());
            test1();
        }
```

```
    // create a static Runnable object using a lambda and use it as the Runnable for a new Thread
    static final Runnable CONSUMER = () -> {
        remove(QUEUE);
    };
    ...
    Thread t = new Thread(CONSUMER);
```

Reply:
I'll pass on the lambdas.